pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -204,5 +261,149 @@ func (c *MizarArktosNetworkController) processNetworkCreation(network *v1.Networ
+	manifestData, err := GetCRDVpcOrSubnetSpec(defaultTemplatePath, vpcOrSubnetName, tenant)
+	if err != nil {
+		klog.Errorf("Mizar-Arktos-Network-controller - at (%s) get JSON spec for CRD (%s) in error: (%v)", defaultTemplatePath, vpcOrSubnetName, err)
+		return err
+	}
+
+	// Create VPC or Subnet object
+	err = createUnstructuredObject([]byte(manifestData), vpcOrSubnetName, discoveryClient, dynamicClient)
+	if err != nil {
+		klog.Errorf("Mizar-Arktos-Network-controller - create CRD object (%s) in error: (%v)", vpcOrSubnetName, err)
+		return err
+	}
+	klog.V(3).Infof("Mizar-Arktos-Network-controller - create CRD object: (%s) successfully", vpcOrSubnetName)
+
+	return nil
+}
+
+func createUnstructuredObject(data []byte, vpcOrSubnetName string, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) error {
+	unstructuredObj := &unstructured.Unstructured{}
+	decUnstructured := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+
+	// Get GVK(Group Version Kind)
+	_, gvk, err := decUnstructured.Decode(data, nil, unstructuredObj)
+	if err != nil {
+		klog.Errorf("Mizar-Arktos-Network-controller: getting GVK in error (%v).", err)
+		return err
+	}
+	klog.V(4).Infof("Mizar-Arktos-Network-controller - get Name : (%s) and GVK: (%s)", unstructuredObj.GetName(), gvk.String())
+
+	// Get mapping from GVK for GVR (Group Version Resource) used by dynamic client resource
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(discoveryClient))
+	klog.V(4).Infof("Mizar-Arktos-Network-controller - Name: %s - GVK group kind : (%v) - GVK version: (%v)", unstructuredObj.GetName(), gvk.GroupKind(), gvk.Version)
+	mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+
+	klog.V(4).Infof("Mizar-Arktos-Network-controller - Name: %s - GVK group kind : (%s) - GVK version: (%s)", unstructuredObj.GetName(), unstructuredObj.GetKind(), unstructuredObj.GetAPIVersion())
+	if err != nil {
+		klog.Errorf("Mizar-Arktos-Network-controller: get mapping between GVK and GVR in error (%v).", err)
+		return err
+	}
+
+	klog.V(4).Infof("Mizar-Arktos-Network-controller - Name: %s - get mapping scope name: (%s) - meta RESTScopeNameNamespace: (%s)", unstructuredObj.GetName(), mapping.Scope.Name(), meta.RESTScopeNameNamespace)
+
+	// Create dynamic client resource
+	var dynamicClientResource dynamic.ResourceInterface
+	if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
+		// namespaced resources should specify the namespace under tenant
+		if unstructuredObj.GetNamespace() == "" {
+			unstructuredObj.SetNamespace("default")
+		}
+		namespace := unstructuredObj.GetNamespace()
+		klog.V(4).Infof("Mizar-Arktos-Network-controller - mapping resource: (%v) - set tenant: (%s) - namespace : (%s)", mapping.Resource, "system", namespace)
+		dynamicClientResource = dynamicClient.Resource(mapping.Resource).NamespaceWithMultiTenancy(namespace, "system")
+
+	} else {
+		// for cluster-wide resources
```

Comment:
when does it  have cluster-wide resource, and this branch is entered?

Reply:
This is the standard checking of the dynamic Client codes - check whether dynamicClientResource is namespaced resources or cluster-wide resources. We just follow it up.

Reply:
This branch is not entered.